daemon/daemon.go

```diff
@@ -353,12 +353,21 @@ func (d *Daemon) sync() jobFunc {
 		if err != nil {
 			return result, err
 		}
-		head, err := d.Repo.Revision(ctx, d.GitConfig.Branch)
-		if err != nil {
+		var latestVerifiedRev string
+		if latestVerifiedRev, _, err = latestValidRevision(ctx, d.Repo, d.GitConfig); err != nil {
 			return result, err
-		}
-		result.Revision = head
-		return result, nil
+		} else if head, err := d.Repo.Revision(ctx, "heads/"+d.GitConfig.Branch); err != nil {
```

Comment:
This is done in a couple of places, I assume because those calls need to unambiguously check a branch head. Could it be put in a method `BranchHead`?